src/main/java/com/esri/samples/map/generate_offline_map_with_local_basemap/GenerateOfflineMapWithLocalBasemap.java

```diff
@@ -0,0 +1,331 @@
+      Scene scene = new Scene(stackPane);
+      scene.getStylesheets().add(getClass().getResource("/css/style.css").toExternalForm());
+
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Generate Offline Map With Local Basemap Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create a map view
+      mapView = new MapView();
+
+      // create a button to take the map offline
+      createTakeMapOfflineButton();
+
+      // create a portal item with the itemId of the web map
+      Portal portal = new Portal("https://www.arcgis.com");
+      PortalItem portalItem = new PortalItem(portal, "acc027394bc84c2fb04d1ed317aac674");
+
+      // create a map with the portal item
+      map = new ArcGISMap(portalItem);
+      addMapLoadedListener();
+      // set the map to the map view
+      mapView.setMap(map);
+      // update the download area box whenever the viewpoint changes
+      mapView.addViewpointChangedListener(viewpointChangedEvent -> updateDownloadArea());
+
+      // create a progress bar to show download progress
+      createProgressIndicator();
+
+      // when the take map offline button is clicked, start the offline map task job
+      offlineMapButton.setOnAction(e -> {
+        try {
+          offlineMapButton.setVisible(false);
+          progressBar.setVisible(true);
+
+          // specify the extent, min scale, and max scale as parameters
+          double minScale = mapView.getMapScale();
+          double maxScale = map.getMaxScale();
+          // minScale must always be larger than maxScale
+          if (minScale <= maxScale) {
+            minScale = maxScale + 1;
+          }
+
+          // create an offline map task with the map
+          task = new OfflineMapTask(map);
+          // create default generate offline map parameters
+          ListenableFuture<GenerateOfflineMapParameters> generateOfflineMapParametersListenableFuture = task.createDefaultGenerateOfflineMapParametersAsync(downloadArea.getGeometry(), minScale, maxScale);
+          // get the offline map parameters from the offline map task
+          generateOfflineMapParameters = generateOfflineMapParametersListenableFuture.get();
+          generateOfflineMapParametersListenableFuture.addDoneListener(() -> {
+
+            // define the directory in which the downloaded .tpk file relevant to this offline map is saved
+            downloadedBasemapSavedLocation = new File("./samples-data/naperville/").getAbsolutePath();
+            // get the name of the basemap file from the offline map as supplied by the map's author (in this instance naperville_imagery.tpk)
+            String referenceBasemapFileName = generateOfflineMapParameters.getReferenceBasemapFilename();
+
+            // if the basemap file name isn't empty, search for that file name within local computer
+            if (!referenceBasemapFileName.isEmpty()) {
```

Comment:
Is it likely this could be empty? Or could it be null (if so, you'd want to check for that)?